src/ecdsa/curves.py

```diff
@@ -166,6 +167,10 @@ def __repr__(self):
     "brainpoolP512r1",
 )
 
+Curve25519Weier = Curve("Curve25519Weier",
+                        ecdsa.curve_25519_weier,
+                        ecdsa.generator_25519_weier,
+                        (1, 3, 101, 110))
```

Comment:
this is not the correct OID, this is an OID for X25519

Reply:
I really do not know what is the actually OID of Weierstrass Form of Curve 25519 (aka wei25519). Please suggest

Reply:
I don't know, that's the first time I've heard of people actually using Curve25519 in Weierstrass form...
Check what OID BouncyCastle uses when you write the key to PKCS#8 file?

Reply:
I really could not find the OID for wei25519. How could I solve this situation?

Reply:
Could you generate the Wei25519 key using BouncyCastle, save it to a PKCS#8 file and post here?

Reply:
-----BEGIN EC PRIVATE KEY-----
MIIBTwIBAQQgDH+gRomUb4CHd+xaBRNBRlpoGbuATwVo+Jk6b8Uy/feggeEwgd4C
AQEwKwYHKoZIzj0BAQIgf////////////////////////////////////////+0w
RAQgKqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqmEkUoUQEIHtCXtCXtCXtCXtC
XtCXtCXtCXtCXtCXtCYLXpx3EMhkBEEEKqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq
qqqqqqqtJFogrhmhuKCGtOAe3Sx3SNFMkj1Nfm18YbIp6cWifs7T2QIgEAAAAAAA
AAAAAAAAAAAAABTe+d6i95zWWBJjGlz10+0CAQihRANCAAQcnP48Ark4a7C6sgBO
Fb2LOY4+StfnnO3Hfx5TqeUhfXhOaTH7A4lMLfMvtIGNapdHM114uHACjRf5Qh2a
OGKp
-----END EC PRIVATE KEY-----

-----BEGIN PUBLIC KEY-----
MIIBMTCB6gYHKoZIzj0CATCB3gIBATArBgcqhkjOPQEBAiB/////////////////
////////////////////////7TBEBCAqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq
qqqYSRShRAQge0Je0Je0Je0Je0Je0Je0Je0Je0Je0Je0JgtenHcQyGQEQQQqqqqq
qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq0kWiCuGaG4oIa04B7dLHdI0UySPU1+
bXxhsinpxaJ+ztPZAiAQAAAAAAAAAAAAAAAAAAAAFN753qL3nNZYEmMaXPXT7QIB
CANCAAQcnP48Ark4a7C6sgBOFb2LOY4+StfnnO3Hfx5TqeUhfXhOaTH7A4lMLfMv
tIGNapdHM114uHACjRf5Qh2aOGKp
-----END PUBLIC KEY-----

Reply:
hmm, so both the private and public key save the curve parameters with the key, they don't use the name/OID; while valid, that's not supported in python-ecdsa: #39 

Is there an api setting in BC to save using named_curve format (instead of the explicit format it used)?

Reply:
I am not sure that I understand your question correctly. Do you mean to embed the curve name in the keys?

Reply:
essentially yes

For ECDSA keys (which mathematically speaking are points on a 2D plane) an application that uses them needs to know on what curve they lay on, that information needs to be encoded in the private and public keys or transferred out of band. When it's included in the files, there are 3 possible encodings: explicit parameters, where the constants that define the curve are spelled out explicitly, as a set of integers; named_curve, where the curve is identified only as an OID, that usually is then transformed into human-readable name, like `secp256r1`; or "implicit", where the curve parameters are supposed to be taken from the CA certificate that signed the public key (I didn't see any software that supports it though)

See https://tools.ietf.org/html/rfc5480#section-2.1.1

in both files the curve parameters are encoded explicitly

A bit of quick googling suggests that the parameters needs to be created using X962NamedCurves.GetOid("prime256v1") : https://stackoverflow.com/a/9432083/462370 but I don't know if you're using .NET or the Java BC

Side note: the "-----BEGIN EC PRIVATE KEY-----" is the legacy, OpenSSL format, PKCS#8 has "-----BEGIN PRIVATE KEY-----" header, that being said both support the different ways of encoding the curve parameters